src/catchup/CatchupRange.cpp

```diff
@@ -37,4 +39,45 @@ operator!=(CatchupRange const& x, CatchupRange const& y)
 {
     return !(x == y);
 }
+
+CatchupRange
+makeCatchupRange(ResolvedCatchupConfiguration const& configuration,
+                 HistoryManager const& historyManager)
+{
+    auto toCheckpoint =
+        historyManager.nextCheckpointLedger(configuration.toLedger() + 1) - 1;
+
+    auto firstNotKnownCheckpoint = [&]() {
+        auto lcl = configuration.lastClosedLedger();
+        auto nextCheckpoint = historyManager.nextCheckpointLedger(lcl + 1) - 1;
+        if (lcl == nextCheckpoint)
+        {
+            return nextCheckpoint + historyManager.getCheckpointFrequency();
+        }
+        return nextCheckpoint;
+    }();
+
+    auto catchupFromLedger =
+        configuration.toLedger() > configuration.count()
+            ? configuration.toLedger() - configuration.count()
+            : 0;
```

Comment:
what does 0 mean here?

Reply:
It means ledger number zero. Probably could use 1 or 2 here as well.

Reply:
yeah I'd rather not use 0 as 0 is not a valid ledger number